xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp

```diff
@@ -229,6 +229,9 @@ bool CWebVTTHandler::CheckSignature(const char* buffer)
 
 void CWebVTTHandler::DecodeLine(std::string line, std::vector<subtitleData>* subList)
 {
+  // We have to remove all \r because it causes the line to display empty box "tofu"
+  StringUtils::Replace(line, "\r", "");
```

Comment:
This will fix the immediate error, but I am concerned that the WebVTT spec Section 6.1 for the parser states
"
Let input be the string being parsed, after conversion to Unicode, and with the following transformations applied:
Replace all U+0000 NULL characters by U+FFFD REPLACEMENT CHARACTERs.
Replace each U+000D CARRIAGE RETURN U+000A LINE FEED (CRLF) character pair by a single U+000A LINE FEED (LF) character.
**Replace all remaining U+000D CARRIAGE RETURN characters by U+000A LINE FEED (LF) characters.**
"
That spec assumes that a file is treated as a single a single string, where here I guess external process has split a file into lines.  So what I am concerned is the case where CARRIAGE RETURN is present in the file stream and not followed by LINE FEED.  In this case it has to be replaced by LINE FEED (not stripped) creating another line.

Reply:
Currently subtitles files formatted with "LF" as return character, not works correctly with Kodi,
there is missing a proper way to handling this type of return type

we should make some changes in how is handled the ReadLine in SubtitleStream.cpp and some other related changes in codec parsers

I think it is better to do this changes separately, for now this fix go well as is
when it will be fixed properly if needed just remove this line